mbed_targets/_internal/target_attribute_hierarchy_parsers/accumulating_attribute_parser.py

```diff
@@ -13,18 +13,31 @@
 from typing import Dict, List, Any, Deque
 
 ACCUMULATING_ATTRIBUTES = ["extra_labels", "macros", "device_has", "features", "components"]
-ALL_ACCUMULATING_ATTRIBUTES = ACCUMULATING_ATTRIBUTES + [
-    "extra_labels_remove",
-    "extra_labels_add",
-    "macros_remove",
-    "macros_add",
-    "device_has_remove",
-    "device_has_add",
-    "features_remove",
-    "features_add",
-    "components_remove",
-    "components_add",
-]
+ACCUMULATING_ATTRIBUTES_MODIFIERS = {"add": "_add", "remove": "_remove"}
+
+
+def _modifying_attribute_name(attribute: str, modifying_action: str) -> str:
+    return f"{attribute}{ACCUMULATING_ATTRIBUTES_MODIFIERS[modifying_action]}"
+
+
+def _list_all_accumulating_attributes_and_modifiers() -> List[str]:
+    """Provides a list of all accumulating attributes and their modifying attributes.
+
+    For example, the accumulating attribute "extra_labels" can be modified by "_add" and
+    "_remove" to make "extra_labels_add" and "extra_labels_remove".
+
+    Returns:
+        A list of all the accumulating attributes and their modifiers.
+    """
+    all_accumulating_attributes = ACCUMULATING_ATTRIBUTES.copy()
+    for modifier_action in ACCUMULATING_ATTRIBUTES_MODIFIERS.keys():
```

Comment:
Not sure we need to call the `keys` here, by default the dictionary iterator should iterate through the keys.

Reply:
45ed7ec